turbopack/crates/turbopack-core/src/ident.rs

```diff
@@ -95,13 +95,18 @@ impl ValueToString for AssetIdent {
             s.push(')');
         }
 
-        if let Some(part) = self.part {
-            let part = part.to_string().await?;
-            // facade is not included in ident as switching between facade and non-facade shouldn't
-            // change the ident
-            if part.as_str() != "facade" {
-                write!(s, " <{}>", part)?;
+        if !self.parts.is_empty() {
+            s.push_str(" (");
+            for (i, part) in self.parts.iter().enumerate() {
+                if i > 0 {
+                    s.push_str(", ");
+                }
+                let part = part.to_string().await?;
+                // Note: facade should be included in ident because the asset type is not
+                // exactly the same.
```

Comment:
Facade and non tree shaken module need to have the same ident for HMR to work correctly. Otherwise switching between both modes has weird effects on HMR.

Facade and non tree shaken module won't exist at the same time since the module content defines if a module is tree shaken or not